pkg/settings/updater.go

```diff
@@ -41,27 +41,50 @@ func EncodeFloat(f float64) string {
 	return strconv.FormatFloat(f, 'E', -1, 64)
 }
 
-// Updater is a helper for updating the in-memory settings.
+// DefaultsUpdater is a helper for updating the in-memory settings.
```

Comment:
But we are, the DefaultsUpdater resets everything that isn't explicitly set to its default value.